tldw_Server_API/app/core/Ingestion_Media_Processing/persistence.py

```
@@ -99,6 +107,28 @@ def _safe_int(value: Any, default: int = 0) -> int:
 
+def _resolve_media_writer(db_instance: Any) -> Any:
+    if hasattr(db_instance, "backend") or hasattr(db_instance, "db_path"):
+        return get_media_repository(db_instance)
+    return db_instance
+
+
+def _with_media_db_session(
+    *,
+    db_path: str,
+    client_id: str,
+    operation: Callable[[MediaDatabase], Any],
```

Ruff (0.15.6)

[error] Undefined name `MediaDatabase`

(F821)